### DISCUSSION TITLE
Inspiration...

### DISCUSSION THREAD

**Discussion by OakdaleFTL** [rxhanson/RectanglePro-Community#240] [General]
Was the old Mac OS 8/9 utility Drop Drawers an inspiration for the stash functionality? (I still use it on my original iMac!.)

**Comment by rxhanson** [comment]
I wish my Mac experience went back that far - my first Mac was an iBook G4 in my 2nd year of college (hence the Tiger wallpaper in the Snap Areas tab). 

Stash is my copy of the excellent Tuck app from Irradiated Software. The author, Steve, was kind enough to give me the go ahead to copy it. What a cool guy!

**Reply by OakdaleFTL** [reply]
Kudos to Steve, as well, then! I still have (and occasionally use) my iBook G4 — running OS X Panther, with the ability to run Mac OS 9 applications, like Glider Pro...

**Comment by OakdaleFTL** [comment]
I've just had an interesting experience: The pop-up notification for a new version prompted me to update the app. Without fearing the consequences, I did so — with three other apps (in different spaces) stashed...one was the Vivaldi browser, one window showing, the rest minimized. When Rectangle Pro relaunched, the browser window had become unreachable!
It did appear when Mission Control was invoked, and was listed as active in the Dock icon's list; but it wouldn't un-stash (nor did Rectangle Pro list it as stashed).
I selected "Unstash All" and quit Rectangle Pro.
Still the browser window remained in _limbo_!

Finally I quit Vivaldi and re-launched it. The lost window returned (positioned as if it had been invoked from its stashed place...) and all was back to normal!
Well, almost: I'd had a Vivaldi Snapshot running in another space, where the new instance of Rectangle Pro had forgot about it too! :) _Rinse-and-repeat_ and all's well and good!

I wouldn't call it a bug... But is this unexpected behavior? :)

**Reply by rxhanson** [reply]
Hmm. Thanks for detailing this. I would call it a bug, as there's a few things that surprisingly didn't happen.

Rectangle Pro is supposed to unstash all when it closes, and it will detect stashed windows upon launch. I'll dig back through and see if there are some edge cases that I've missed. It is likely that windows stashed on other spaces might not be unstashed properly. 

macOS restricts window placements so that they actually cannot be moved off of the screen entirely, so if this happens again, you should be able to drag resize a window by a 1px edge on the edge of the screen.
